File: _03_JPA/projects/_03_data_jpa/src/main/java/inf/datajpa/controller/MemberController.java
package inf.datajpa.controller;

import javax.annotation.PostConstruct;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.web.PageableDefault;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import inf.datajpa.dto.MemberDto;
import inf.datajpa.entity.Member;
import inf.datajpa.repository.MemberRepository;
import lombok.RequiredArgsConstructor;

@RestController
@RequiredArgsConstructor
public class MemberController {

	private final MemberRepository memberRepository;

	@GetMapping("/members")
	public Page<MemberDto> list(@PageableDefault(size = 5) Pageable pageable) {
		return memberRepository.findAll(pageable)
			.map(MemberDto::new);
	}

	@PostConstruct
	public void init() {
		for (int i = 0; i < 100; i++) {
			memberRepository.save(new Member("user" + i, i));
		}
	}

}

File: BackEnd/Kotlin/projects/kt-security-example/src/main/kotlin/com/example/ktsecurityexample/member/controller/MemberController.kt
package com.example.ktsecurityexample.member.controller

import com.example.ktsecurityexample.member.dto.MemberDtoRequest
import com.example.ktsecurityexample.member.entity.Member
import com.example.ktsecurityexample.member.service.MemberService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController


@RestController
class MemberController(
    private val memberService: MemberService,
) {
    /**
     * 회원가입
     */
    @PostMapping("/api/members/sign-up")
    fun signUp(
        @RequestBody memberDtoRequest: MemberDtoRequest,
    ): ResponseEntity<Member> {
        val member = memberService.signUp(memberDtoRequest)
        return ResponseEntity.ok(member)
    }
}
